circuit_knitting/cutting/cutqc/wire_cutting_dynamic_definition.py

```diff
@@ -0,0 +1,524 @@
+                        subcircuit_idx
+                    ]
+                    for subcircuit_qubit_idx, qubit_state in enumerate(
+                        subcircuit_state
+                    ):
+                        qubit_idx = subcircuit_out_qubits[subcircuit_idx][
+                            subcircuit_qubit_idx
+                        ]
+                        if qubit_state == "active":
+                            binary_full_state[qubit_idx] = binary_bin_id[0]
+                            binary_bin_id = binary_bin_id[1:]
+                        else:
+                            binary_full_state[qubit_idx] = "%s" % qubit_state
+                #print("reordered qubit state = {}".format(binary_full_state))
+                merged_qubit_indices = []
+                for qubit, qubit_state in enumerate(binary_full_state):
+                    if qubit_state == "merged":
+                        merged_qubit_indices.append(qubit)
+                num_merged = len(merged_qubit_indices)
+                average_state_prob = bin_prob / 2**num_merged
+                for binary_merged_state in itertools.product(
+                    ["0", "1"], repeat=num_merged
+                ):
+                    for merged_qubit_ctr in range(num_merged):
+                        binary_full_state[
+                            merged_qubit_indices[merged_qubit_ctr]
+                        ] = binary_merged_state[merged_qubit_ctr]
+                    full_state = "".join(binary_full_state)[::-1]
+                    full_state_idx = int(full_state, 2)
+                    reconstructed_prob[full_state_idx] = average_state_prob
+                    #print(
+                    #    "--> full state {} {:d}. p = {:.3e}".format(
+                    #        full_state, full_state_idx, average_state_prob
+                    #    )
+                    #)
+                #print()
+    return reconstructed_prob
+
+
+def get_reconstruction_qubit_order(
+    full_circuit: QuantumCircuit,
+    complete_path_map: dict[Qubit, list[dict[str, int | Qubit]]],
+    subcircuits: list[QuantumCircuit],
+) -> dict[int, list[int]]:
+    """
+    Get the output qubit in the full circuit for each subcircuit
+
+    Args:
+        full_circuit: The original quantum circuit
+        complete_path_map: The complete path through the subcircuits
+        subcircuits: The subcircuits
+    Returns:
+        The output qubit for the original circuit
+    """
+    subcircuit_out_qubits: dict[int, list[int]] = {
+        subcircuit_idx: [] for subcircuit_idx in range(len(subcircuits))
+    }
+    for input_qubit in complete_path_map:
+        path = complete_path_map[input_qubit]
+        output_qubit = path[-1]
+        subcircuit_out_qubits[output_qubit["subcircuit_idx"]].append(
+            (
+                output_qubit["subcircuit_qubit"],
+                full_circuit.qubits.index(input_qubit),
+            )  # type:ignore
```

Comment:
```suggestion
            )  # type:ignore
```

Any comment on the 3 ignored types here? We can't get them to work with our type hints?

Reply:
I checked the type and updated a correct annotation.